src/cascadia/TerminalApp/AppActionHandlers.cpp

```diff
@@ -844,18 +844,54 @@ namespace winrt::TerminalApp::implementation
         }
 
         _UpdateTeachingTipTheme(WindowRenamer().try_as<winrt::Windows::UI::Xaml::FrameworkElement>());
-        WindowRenamer().IsOpen(true);
 
-        // PAIN: We can't immediately focus the textbox in the TeachingTip. It's
-        // not technically focusable until it is opened. However, it doesn't
-        // provide an event to tell us when it is opened. That's tracked in
-        // microsoft/microsoft-ui-xaml#1607. So for now, the user _needs_ to
-        // click on the text box manually.
+        // BODGY: GH#12021
+        //
+        // TeachingTip doesn't provide an Opened event.
+        // (microsoft/microsoft-ui-xaml#1607). But we want to focus the renamer
+        // text box when it's opened. We can't do that immediately, the TextBox
+        // technically isn't in the visual tree yet. We have to wait for it to
+        // get added some time after we call IsOpen. How do we do that reliably?
+        // Usually, for this kind of thing, we'd just use a one-off
+        // LayoutUpdated event, as a notification that the TextBox was added to
+        // the tree. HOWEVER:
+        //   * The _first_ time this is fired, when the box is _first_ opened,
+        //     tossing focus doesn't work on the first LayoutUpdated. It does
+        //     work on the second LayoutUpdated. Okay, so we'll wait for two
+        //     LayoutUpdated events, and focus on the second.
+        //   * On subsequent opens: We only ever get a single LayoutUpdated.
+        //     Period. But, you can successfully focus it on that LayoutUpdated.
+        //
+        // So, we'll keep track of how many LayoutUpdated's we've _ever_ gotten.
+        // If we've had at least 2, then we can focus the text box.
         //
         // We're also not using a ContentDialog for this, because in Xaml
         // Islands a text box in a ContentDialog won't receive _any_ keypresses.
         // Fun!
         // WindowRenamerTextBox().Focus(FocusState::Programmatic);
+        _renamerLayoutUpdatedRevoker.revoke();
+        _renamerLayoutUpdatedRevoker = WindowRenamerTextBox().LayoutUpdated(winrt::auto_revoke, [weakThis = get_weak()](auto&&, auto&&) {

```

Comment:
i believe assigning over this thing _does_ call revoke

Reply:
I figured, but who knows how often we're burned by that